bindings/python/samples/runtext.py

```diff
@@ -9,23 +9,27 @@ class RunText(SampleBase):
     def __init__(self, *args, **kwargs):
         super(RunText, self).__init__(*args, **kwargs)
         self.parser.add_argument("-t", "--text", help="The text to scroll on the RGB LED panel", default="Hello world!")
+        self.parser.add_argument("-s", "--speed", help="Approximate letters per second.", default="-s0",type=float)
+        #self.parser.add_argument("--led-panel-type", action="store", help="Needed to initialize special panels. Supported: 'FM6126A'", default="", type=str)
 
-    def run(self):
+    def run(self, s):
         offscreen_canvas = self.matrix.CreateFrameCanvas()
         font = graphics.Font()
         font.LoadFont("../../../fonts/7x13.bdf")
         textColor = graphics.Color(255, 255, 0)
         pos = offscreen_canvas.width
         my_text = self.args.text
 
-        while True:
+        #while True:
+        if(s > 0.0):
```

Comment:
What is the purpose of this change ?

Reply:
I plan on calling the run function based on different scenarios, so I want to control the speed of the text - ranging from 0 and greater than 0.

Reply:
the run function is the function running in an eternal loop in the thread. If you want to change that, you probably want to rename the funcion.

Reply:
How about I just write a new function that take a boolean value to perform the while loop like this 
```
    def runTextWithSpeed(self, speedOfText, textToDisplay, textColorToDisplay, displaySomething):
        while(displaySomething):
            offscreen_canvas = self.matrix.CreateFrameCanvas()
            font = graphics.Font()
            font.LoadFont("../../../fonts/7x13.bdf")
            #textColor = graphics.Color(255, 255, 0)
            textColor = textColorToDisplay
            pos = offscreen_canvas.width
            thisText = textToDisplay

            if(speedOfText == 0):
                pos = 20
                offscreen_canvas.Clear()
                graphics.DrawText(offscreen_canvas, font, pos, 10, textColor, thisText)
            elif(speedOfText > 0):
                offscreen_canvas.Clear()
                len = graphics.DrawText(offscreen_canvas, font, pos, 10, textColor, thisText)
                pos -= 1
                if (pos + len < 0):
                    pos = offscreen_canvas.width
                time.sleep(1.0/speedOfText)
                offscreen_canvas = self.matrix.SwapOnVSync(offscreen_canvas)
        else:
            offscreen_canvas.Clear()
```

Reply:
here, you're creating a frame canvas and load the font in every loop iteration. You'll run out of memory soon.

Reply:
Ok I see that -- -I've modified the function to this:
```
    def runTextWithSpeed(self, speedOfText, textToDisplay, textColorToDisplay, displaySomething):
        if(displaySomething):
            offscreen_canvas = self.matrix.CreateFrameCanvas()
            font = graphics.Font()
            font.LoadFont("../../../fonts/7x13.bdf")
            #textColor = graphics.Color(255, 255, 0)
            textColor = textColorToDisplay
            pos = offscreen_canvas.width
            thisText = textToDisplay

            if(speedOfText == 0):
                pos = 20
                offscreen_canvas.Clear()
                graphics.DrawText(offscreen_canvas, font, pos, 10, textColor, thisText)
            elif(speedOfText > 0):
                while True:
                    offscreen_canvas.Clear()
                    len = graphics.DrawText(offscreen_canvas, font, pos, 10, textColor, thisText)
                    pos -= 1
                    if (pos + len < 0):
                        pos = offscreen_canvas.width
                    time.sleep(1.0/speedOfText)
                    offscreen_canvas = self.matrix.SwapOnVSync(offscreen_canvas)
        else:
            offscreen_canvas.Clear()
```

Reply:
Hi Henner,
Also when I create a an offscreen_canvas, Do I need to pass in the number of rows and columns as parameters ? I traced the it back to the "led-matrix.cc" where frame-buffer arguments are passed in. 
Thanks

Reply:
The matrix creates the frame canvas internally, so it will return you an instance with everything properly set.

Reply:
Thx  -- Well if  my mofifed code looks OK to you then I suppose there is no need to proceed further with this pull request because I can simply add this function to my project.

Reply:
Perfect!
Have fun with your project and LEDs!